src/Computation/Values/And.cs

```diff
@@ -33,17 +34,22 @@ private static IValue ShortCircuit(IValue left, ConstantUnsigned right)
             ? right
             : right.Not().IsZero
                 ? left
-                : new And(left, right);
+                : Create(left, right);
+    }
+
+    private static IValue Create(IValue left, ConstantUnsigned right)
```

Comment:
The reason for this refactor is so that when I add the associativity changes I can just change this method to be:

```csharp
return left is IConstantValue cl
    ? cl.AsUnsigned().And(right)
    : left is And l
        ? new And(l._left, Create(l._right, right))
        : new And(left, right);
```

This way the recursive part only affects this method alone rather than it constantly calling `ShortCircuit`, it's a minor optimisation, but it also just felt more correct and simpler to comprehend. The `public Create` method just checks if either is a constant and then attempts to short circuit based on the right one definitely being a constant. If that falls through then it calls the `private Create` which is responsible for attempting to merge together constant values by either doing the constant operation if both are constant or pushing the single constant value as far to the right as it can until it either encounters another constant or hits the end of the `Or` train.

Reply:
Logic seems fine.